plugins/destination/mysql/client/schema.go

```diff
@@ -165,7 +177,9 @@ func (c *Client) createTable(ctx context.Context, table *schema.Table) error {
 			sqlType := arrowTypeToMySqlStr(column.Type)
 			if sqlType == "blob" || sqlType == "text" {
 				// `blob/text` SQL types require specifying prefix length to use for the primary key
-				builder.WriteString("(64)")
+				// https://dev.mysql.com/doc/refman/8.0/en/innodb-limits.html
+				// The index key prefix length limit is 767 bytes for InnoDB tables that use the REDUNDANT or COMPACT row format. For example, you might hit this limit with a column prefix index of more than 191 characters on a TEXT or VARCHAR column, assuming a utf8mb4 character set and the maximum of 4 bytes for each character.
+				builder.WriteString("(" + fmt.Sprintf("%d", maxPrefixLength) + ")")
```

Comment:
```suggestion
				builder.WriteString("(" + strconv.Itoa(maxPrefixLength) + ")")
```

fmt is costly, use itoa directly